### ISSUE TITLE
How to use event row-clicked <b-table>.

### ISSUE THREAD

**Issue by vitorvr** [bootstrap-vue/bootstrap-vue#774]
Hi guys, I'm trying to use event row-clicked to fill some props when click on a b-table row.
Tried to find an example on google but without success.
How can I use this event?
I'm currently using a button for this, but I want to change it.

`      <b-table bordered striped hover :items="environments" :fields="fields" :current-page="currentPage" :per-page="perPage" :filter="filter">
<template slot="actions" scope="environment">
<b-btn size="sm" @click="log(environment.item)">Details</b-btn>
</template>
</b-table>`

Is it possible use row-clicked or even double click for that? Thanks!

**Comment by tmorehouse** [comment]
```html
<b-table bordered striped hover 
         :items="environments"
         :fields="fields"
         :current-page="currentPage"
         :per-page="perPage"
         :filter="filter"
         v-model="shownItems"
         @row-clicked="myRowClickHandler"
>
  <template slot="actions" scope="environment">
    <b-btn size="sm" @click="log(environment.item)">Details</b-btn>
  </template>
</b-table>
```

```js
methods: {
  myRowClickHandler(record, index) {
    // 'record' will be the row data from items
    // `index` will be the visible row number (available in the v-model 'shownItems')
    log(record); // This will be the item data for the row
  }
}
```

**Comment by tmorehouse** [comment]
Adding `dblclick` support to rows might also be a handy feature for us to add to `b-table`

**Comment by vitorvr** [comment]
I missed the @ before the event! :(
Thanks so much @tmorehouse! I appreciate your help!
`click` is enough.

Thanks!

**Comment by PierBover** [comment]
Strange that an example of using `row-clicked` isn't in the docs...

https://bootstrap-vue.js.org/docs/components/table/

@tmorehouse would you accept a PR?

**Comment by hrakotom** [comment]
Hi,
Sry if this is obvious, but i may have missed it:
- How do you refer to row in a row-clicked handler ( for example to show row details ? )

**Comment by KappaLambda** [comment]
@andthomas I am not sure I got your example to @hrakotom question? How exactly to me use the `row.toggleDetails` function instead of our handler.
example:
```html
<b-table
    @row-clicked="myHandler"  // how instead of my Handler can I use the row.toggleDetails?
>
</b-table>
```

All I want is to show row details when I click the row and not with a button.

**Comment by tmorehouse** [comment]
The row clicked handler is passed a copy of the row's data (item) as the first argument to the handler.

You can then set the value of item._showDetails to true:

```
  // To toggle:
  this.$set(item, '_showDetails', !item._showDetails)
  // or to just show:
  this.$set(item, '_showDetails', true)
  // or to just hide:
  this.$set(item, '_showDetails', false)
```

Using `this.set(...)` ensures that Vue's reactivity will detect teh change (in case the `_showDetails` property doesn't exist in the item)

**Comment by jessie8910** [comment]
Is there any chance I can pass my own parameter to rowclickhandler, since rowclickhandler is a method, and the default parameters are rowItem, index and event. Can I pass my own parameter to this method?
Something like this:
                             <b-tabs card>
                                        <b-tab v-for="comp in components" :key="comp" :title="comp.name">
                                            <b-table show-empty bordered striped hover
                                                        :items="comp.items"
                                                        :fields="comp.fields"
                                                        @row-dblclicked="rowDblClickHandler({comp.name})">
                                            </b-table>
                                        </b-tab>
                              </b-tabs>

**Comment by RudyOnRails** [comment]
@jessie8910 I'm trying to do the same right now. if i figure out solution, i will let you know

**Comment by RudyOnRails** [comment]
@jessie8910 looks as if you pass in your own arguments, you can control the arguments that are yielded. NOt sure that makes sense but best i could explain in js.

`@row-clicked=`pbrModal("edit", data.item)`)`

**Comment by tmorehouse** [comment]
@jessie8910 

You could add the component name to your items data (just don't include it in the fields passed to b-table):

```html
<b-tabs card>
  <b-tab v-for="comp in components" :key="comp" :title="comp.name">
    <b-table
      show-empty bordered striped hover
      :items="comp.items.map(item => ({_name; comp.name, ...item}))"
      :fields="comp.fields"
      @row-dblclicked="rowDblClickHandler">
    </b-table>
  </b-tab>
</b-tabs>
```

Now the `item` record passed to the row double click handler will include the field `_name`.

**Comment by jessie8910** [comment]
@tmorehouse 
That is a nice way to do it, thank you, it works!

**Comment by bmarkovic** [comment]
@jessie8910  You could directly declare functions as event handlers to avoid polluting `items` with potentially huge number of copies of the same string:

```vue
<b-tabs card>
  <b-tab v-for="comp in components" :key="comp.name" :title="comp.name">
    <b-table
      show-empty bordered striped hover
      :fields="comp.fields"
      @row-dblclicked="(item, index, event) => rowDblClickHandler(comp.name, item, index, event)">
    </b-table>
  </b-tab>
</b-tabs>
```

**Comment by PierBover** [comment]
@bmarkovic doesn't that create a function every time an item is rendered?

**Comment by tmorehouse** [comment]
@PierBover It would create one function for each table rendered

**Comment by PierBover** [comment]
Ah sorry, brain fart.

**Comment by thomporter** [comment]
Just thought I'd throw this in here.  

I have a selectable `b-table` (mode=single) inside of a `b-tab`.  When users select a row, some information shows to the right of the table, including a button that lets them open a new `b-tab` to act on that record (it's a ton of data, so each row they work on gets it's own tab.)  The issue I was having was when you go back to the "list" tab the selected row was lost.  I was using `v-if` and didn't think to just change it to `v-show` at first.

I first tried solving it with a watcher, but this did not work, wondering if anyone knows why:

```javascript
watch: {
  activeTab(n) {
    if (n == 'list' && this.selectedIndex) { 
      this.$nextTick(function () {
      console.log(this.$refs.selectableTable) // works, I can see the vue component here.
      console.log(this.selectedIndex) // works, gives me the right index # of the row that was previously selected.
      this.$refs.selectableTable.selectRow(this.selectedIndex) // does not do anything.  No row is selected, and no errors in the console.
    })
  }
}
```

Again, I solved the problem by just changing the `v-if` to a `v-show` and now don't have to watch anything, but just wondering why the above wouldn't work.
